src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java

```diff
@@ -141,21 +151,113 @@ StageWithCaseData getActiveStage(UUID caseUUID, UUID stageUUID) {
-        caseDataService.updateCaseData(caseData, stage.getUuid(), Map.of(CaseworkConstants.CURRENT_STAGE, stageType));
-        auditClient.createStage(stage);
         log.info("Created Stage: {}, Type: {}, Case: {}, event: {}", stage.getUuid(), stage.getStageType(), stage.getCaseUUID(), value(EVENT, STAGE_CREATED));
-        notifyClient.sendTeamEmail(caseUUID, stage.getUuid(), teamUUID, caseData.getReference(), emailType);
+
+        StageWithCaseData stageWithCaseData = getStageWithCaseData(caseUUID, stage.getUuid());
+        caseDataService.updateCaseData(caseData, stage.getUuid(), Map.of(CaseworkConstants.CURRENT_STAGE, stage.getStageType()));
+
+        // Update team and user to activate new stage
+        assignTeamAndMemberUserToStage(stageWithCaseData, currentUserUUID);
+
+        // remove old active stage
+        if (activeStageToBeDeactivated != null) {
+            deactivateCurrentActiveStage(caseUUID, activeStageToBeDeactivated);
+        }
+
+        // Update audit and timeline - positioned here maintains timeline consistency.
+        auditClient.createStage(stageWithCaseData);
+
+        // Issue assignment notifications
+        updateAssignmentAudit(stageWithCaseData);
+        sendAssignmentNotifications(stageWithCaseData, currentUserUUID);
+
         return stage;
     }
 
+    private void deactivateCurrentActiveStage(UUID caseUUID, StageWithCaseData activeStage) {
+
+        log.info("Active stage {} exists; removing team allocation to complete stage.", activeStage);
+        updateStageTeam(caseUUID,activeStage.getUuid(), null, null);
+    }
+
+    private StageWithCaseData getCurrentActiveStageWithCaseData(UUID caseUUID) {
+        Set<StageWithCaseData> activeStages = stageRepository.findAllActiveByCaseUUID(caseUUID);
+        if (activeStages.size() > 1) {
+            log.error("Multiple Active Stages discovered for case: {}, stages: {}", caseUUID, activeStages);
+            throw new ApplicationExceptions.MultipleActiveStageException(caseUUID.toString());
+        }
+        ArrayList<StageWithCaseData> listOfActiveStages = new ArrayList<>(activeStages);
+
+        return !listOfActiveStages.isEmpty() ? listOfActiveStages.get(0) : null;
+    }
+
+    private void assignTeamAndMemberUserToStage(StageWithCaseData stage, UUID currentUserUUID) {
+
+        if (stage.getTeamUUID() == null) {
+            log.debug("Updating Team for Stage: {} using default team",stage.getUuid());
+            UUID teamUUID = infoClient.getTeamForStageType(stage.getStageType());
+            stage.setTeam(teamUUID);
+        } else {
+            log.debug("Updating Team for Stage: {} with requested team: {}",stage.getUuid(),stage.getTeamUUID());
+        }
+
+        if (stage.getTeamUUID() != null && stage.getUserUUID() != null) {
+            log.debug("Updating User: {} for Stage: {}",stage.getUserUUID() , stage.getUuid());
+
+            UserDto userInTeam = infoClient.getUserForTeam(stage.getTeamUUID(), stage.getUserUUID());
```

Comment:
Appeciate this is existing code that has been moved across, does it sit better extracted out into the UserPermisisonsService? Happy to raise a ticket.

Reply:
I think that would make the context of the request something different. We aren't checking whether the current user has permissions, only that the user they have allocated this case to, is a member of the required team, this is not necessarily the currently acting user.